tests/test_indexes.py

```diff
@@ -24,6 +24,15 @@
 )
 
 if TYPE_CHECKING:
+    MYPY_CHECKING: bool = True
+    if MYPY_CHECKING:
+        from typing import Any
+
+        from typing_extensions import TypeAlias
+
+        SupportsRichComparison: TypeAlias = Any
+    else:
+        from _typeshed import SupportsRichComparison  # noqa: F401
```

Comment:
Do you mind referencing an issue or describing the discrepancy? Is it just that mypy has an older version of typestubs?

Reply:
Added comments in https://github.com/pandas-dev/pandas-stubs/pull/759/commits/5310eb682334877ae00a06b80082fc80f7467aee

Reply:
> Does this also work for `poe mypy --mypy_nightly`?

Looks like it did!

We may want to use this pattern (`MYPY_CHECKING`) elsewhere when `mypy` and `pyright` make different inferences.  Can't put it in the stubs themselves - that would require users to define `MYPY_CHECKING` as `false` if using pyright, but we can use it in our tests.